distributed/shuffle/_scheduler_plugin.py

```diff
@@ -115,7 +117,13 @@ def __init__(self, scheduler: Scheduler):
         self.heartbeats = defaultdict(lambda: defaultdict(dict))
         self.states = {}
         self.erred_shuffles = {}
-        self.scheduler.add_plugin(self)
+        self.scheduler.add_plugin(self, name="shuffle")
+
+    async def start(self, scheduler: Scheduler) -> None:
+        worker_plugin = ShuffleWorkerPlugin()
+        await self.scheduler.register_worker_plugin(
+            None, dumps(worker_plugin), name="shuffle"
+        )
```

Comment:
So this is the magic that ensures things are cleaned up, because the client will teardown the scheduler which tears down all the worker plugins?

Reply:
And pretty much everything else is downstream renaming changes....

Reply:
Pretty much, yes. IIUC, extensions don't get cleaned up at all. Worker plugins will get torn down on worker close, but for that the method has to be called `teardown` (as seen below). Similarly, we need to move some of the initialization to `setup`.